### PR TITLE
🐛Fix `data-amp-replace` comparison logic

### LINKED ISSUE #19061: Breaking change with amp-list and CORS variable substitution in AMP 1810301740350
## What's the issue?

A change was introduce which updated how the amp-list handles CORS variable replacements. This causes an issue where an error is thrown and the amp-list can no longer hydrate.

`Error fetching amp-list: URL variable substitutions in CORS fetches from dynamic URLs (e.g. via amp-bind) require opt-in. Please add data-amp-replace="CLIENT_ID(ghs-amp-pages)" to the <AMP-LIST> element.`

Previously this was handled with 

`data-amp-replace="CLIENT_ID" >`

## How do we reproduce the issue?

jsbin

https://jsbin.com/nilijuh/4/edit?html,output

Gist
https://gist.github.com/mikeswright49/9c490ec8f77d10e4d7defa05da5a47d8

1. Update amp-list binding to ="CLIENT_ID"
2. Type into the input box
3. Validate issue in console
4. Update amp=list binding to ="CLIENT_ID(a-key)"
5. Type into input box
6. Reset to validate issue gone

## What browsers are affected?

All browsers

## Which AMP version is affected?

Powered by AMP ⚡ HTML – Version 1810301740350

### PR DESCRIPTION
**Pull request by calebcordry**
Fixes #19061 

The new expander had inadvertently introduced a new API that `UrlReplacements.collectUnwhitelistedVarsSync_` had become dependent on. When the API was changed back in #18554 it broke the intended functionality of this method.

This removes the arguments collected by the `collectVars` object before comparing them to the given `data-amp-replace` options.

### CODE REVIEW

**Comment by calebcordry** [comment]
Working on writing test now, but wanted to get input on the fix. Looks like this test might have caught it but was skipped :(
https://github.com/ampproject/amphtml/blob/3a29c543fca5d5e3eed147520e96f51070956b6e/test/functional/test-url-replacements.js#L1311

File: src/service/url-replacements-impl.js [hunk 1]
```diff
@@ -1049,7 +1049,9 @@ export class UrlReplacements {
     const url = element.getAttribute('src');
     const vars = Object.create(null);
     this.expandStringSync(url, /* opt_bindings */ undefined, vars);
-    const varNames = Object.keys(vars);
+    const varNames = Object.keys(vars)
+        // Strip arguments from collect_vars to match whitelist api.
+        .map(key => key.substring(0, key.indexOf('(')));
```
Comment (choumx) [inline src/service/url-replacements-impl.js]: Seems a bit hacky?

Comment (calebcordry) [reply src/service/url-replacements-impl.js]: I agree. A larger refactor is needed to be less so. Wasn't sure if we should patch and clean up after.

Comment (lannka) [reply src/service/url-replacements-impl.js]: indeed. is it possible to have spaces before `(`?

chatted offline, a better fix would redefine this `collectUnwhitelistedVarsSync` method, and move whitelist logic to the caller. 

I'm fine with a quick fix like this to patch prod.

Comment (calebcordry) [reply src/service/url-replacements-impl.js]: Should not be possible to have spaces before arguments.

Comment (lannka) [reply src/service/url-replacements-impl.js]: I thought we allow `/path/CLIENT_ID (xxx)` ? or did it get normalized somewhere down the path?

Comment (calebcordry) [reply src/service/url-replacements-impl.js]: We construct these in `maybeCollectVars_` with the macro name coming from the macro definition, not the url.

Comment (lannka) [reply src/service/url-replacements-impl.js]: oh, that's good.

**Pull request merged**